docs/nx-cloud/reference/launch-templates.md

```diff
@@ -1,42 +1,296 @@
+The token name is not important here, `MY_REGISTRY_TOKEN` is to demonstrate where the value can be placed.
+
+Reference the npm docs on [how to use a `.npmrc` file](https://docs.npmjs.com/cli/v10/configuring-npm/npmrc)
+{% /callout %}
+
+## Custom Node Version
+
+Nx Agents come with node LTS installed. If you want to use a different version, you can create a custom step to install the desired node version. Here is a simplified launch template using [nvm](https://github.com/nvm-sh/nvm) to install a different node version.
+
+```yaml {% fileName=".nx/workflows/agents.yaml" %}
+launch-templates:
+  node-18:
+    resourceClass: 'docker_linux_amd64/medium'
+    image: 'ubuntu22.04-node20.11-v6'
+    init-steps:
+      - name: Checkout
+        uses: 'nrwl/nx-cloud-workflows/v3.6/workflow-steps/checkout/main.yaml'
+      - name: Install nvm
+        script: |
+          # run nvm install script
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+          # source the updated profile to get the nvm command available
+          source ~/.profile
+          # install the needed version of node via nvm
+          nvm install 21.7.3
+          # echo the current path (which now includes the nvm-provided version of node) to NX_CLOUD_ENV
+          echo "PATH=$PATH" >> $NX_CLOUD_ENV
+      - name: Print node version
+        # confirm that the node version has changed
+        script: node -v
+      - name: Install Node Modules
+        uses: 'nrwl/nx-cloud-workflows/v3.6/workflow-steps/install-node-modules/main.yaml'
+      # Continue setup steps as needed
+```
+
+## Installing various packages
+
+You can use `apt` to install popular linux packages. This is helpful in streamlining setting up various toolchains needed for your workspace.
+
+For example, you can install the [GitHub CLI](https://cli.github.com/) on the agents if needed.
+
+```yaml {% fileName="./nx/workflows/agents.yaml" %}
+launch-templates:
+  my-linux-medium-js:
+    resourceClass: 'docker_linux_amd64/medium'
+    image: 'ubuntu22.04-node20.11-v6'
+    init-steps:
+      - name: Install GH CLI
+        script: |
+          sudo apt install gh -y
+```
+
+{% callout type="note" title="Installing without apt" %}
+If you're trying to install a package that isn't available on `apt`, check that packages install steps for Debian base linux. Usually there are a handful of installation scripts that can be used [similar to `nvm`](#custom-node-version)
+{% /callout %}
+
+## Dynamic Changesets
+
+Nx Agents support dynamically distributing a given changeset for a CI run with a different number of agents and templates.
+This is equivalent of changing the value passed to `--distribute-on` based on your changes. Instead of defining a static number of agents for each change.
+
+Currently, you can define a `small`, `medium` and `large` changeset for Nx Agents to use.
+These sizes are based on the ratio of affected projects Nx Cloud calculates for your PR.
```

Comment:
```suggestion
NxCloud can calculate how big your pull request is based on how many projects in your workspace it affects. You can then configure Nx Agents to dynamically use a different number of agents based on your changeset size.

Here we define a `small`, `medium` and `large` distribution strategy:
```

Reply:
you don't need to use my suggestion, was just trying to see if we could shorten the explanation